internal/cluster/transition.go

```diff
@@ -195,28 +219,34 @@ func (th *transitionHandler) hasClusterCompleteInstallation(sw stateswitch.State
 	return isComplete, nil
 }
 
-func getClusterMonitoringOperatorsStatus(cluster *common.Cluster) (bool, map[models.OperatorStatus][]string) {
-	numberOfBuiltInOperators := 0
-
-	operatorsStatuses := map[models.OperatorStatus][]string{
-		models.OperatorStatusAvailable:   {},
-		models.OperatorStatusProgressing: {},
-		models.OperatorStatusFailed:      {},
+func getClusterMonitoringOperatorsStatus(cluster *common.Cluster) (bool, MonitoredOperatorStatuses) {
+	operatorsStatuses := MonitoredOperatorStatuses{
+		models.OperatorTypeOlm: {
+			models.OperatorStatusAvailable:   {},
+			models.OperatorStatusProgressing: {},
+			models.OperatorStatusFailed:      {},
+		},
+		models.OperatorTypeBuiltin: {
+			models.OperatorStatusAvailable:   {},
+			models.OperatorStatusProgressing: {},
+			models.OperatorStatusFailed:      {},
+		},
 	}
 
 	for _, operator := range cluster.MonitoredOperators {
-		if operator.OperatorType == models.OperatorTypeOlm {
-			// TODO: MGMT-3368
-			// Should change cluster state to be degraded
-			continue
-		}
+		operatorsStatuses[operator.OperatorType][operator.Status] = append(operatorsStatuses[operator.OperatorType][operator.Status], operator.Name)
+	}
 
-		numberOfBuiltInOperators++
+	return len(operatorsStatuses[models.OperatorTypeBuiltin][models.OperatorStatusAvailable]) == countOperators(operatorsStatuses[models.OperatorTypeBuiltin]), operatorsStatuses
+}
 
-		operatorsStatuses[operator.Status] = append(operatorsStatuses[operator.Status], operator.Name)
+func countOperators(in map[models.OperatorStatus][]string) int {
```

Comment:
what it is counting?

Reply:
A total number of operators in all the statuses.